ci/deps/actions-311.yaml

```diff
@@ -0,0 +1,56 @@
+name: pandas-dev
+channels:
+  - conda-forge
+dependencies:
+  - python=3.11
+
+  # build dependencies
+  - versioneer[toml]
+  - cython>=0.29.32
+
+  # test dependencies
+  - pytest>=7.0
+  - pytest-cov
+  - pytest-xdist>=2.2.0
+  - psutil
+  - pytest-asyncio>=0.17
+  - boto3
+
+  # required dependencies
+  - python-dateutil
+  - numpy<1.24
```

Comment:
Can this be unpinned?

Reply:
Let's check, copied 3.10 when I started but seems to work there now

Reply:
Have to keep it pinned. Numba pins it indirectly on the other jobs, but since we don't install it here, we have to keep the pin